### PR TITLE
swap out elasticsearchContainer for opensearchContainer

### PR DESCRIPTION
**Pull request by fredrikpf**

### CODE REVIEW

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java [hunk 1]
```diff
@@ -24,20 +9,37 @@
...
+
+import static no.unit.nva.search.SearchClient.*;
+import static no.unit.nva.search.constants.ApplicationConstants.objectMapperWithEmpty;
+import static no.unit.nva.testutils.RandomDataGenerator.randomUri;
+import static nva.commons.core.ioutils.IoUtils.inputStreamFromResources;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
 @Testcontainers
 public class ElasticsearchTest {
```
Comment (sondrev) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:41]: Shuld we rename the file to OpensearchTest?

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java [hunk 2]
```diff
@@ -47,41 +49,83 @@ public class ElasticsearchTest {
...
+    @Test
+    void canConnectToContainer() throws IOException, InterruptedException{
+
+        var port9200 = container.getMappedPort(9200);
+        var host = container.getHost();
+        var httpHostAddress = host + ":" + port9200;
+
+        HttpClient httpClient = HttpClient.newBuilder().build();
+
+        String authenticatorValue = "admin:admin";
```
Comment (sondrev) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:101]: Can this change? If not it should be final, and perhaps extracted as a constant

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java [hunk 3]
```diff
@@ -47,41 +49,83 @@ public class ElasticsearchTest {
...
+            .withEnv("plugins.security.disabled", "true")
+            .withEnv("discovery.type", "single-node");
+
+    @SuppressWarnings("unchecked")
     @BeforeEach
     void setUp() {
-        RestClientBuilder restClientBuilder = RestClient.builder(HttpHost.create(container.getHttpHostAddress()));
+
+        container.setPortBindings(List.of("9200:9200", "9600:9600"));
+        container.setExposedPorts(List.of(9200, 9600));
```
Comment (sondrev) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:72]: All the ports should be stored as constants

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java [hunk 4]
```diff
@@ -47,41 +49,83 @@ public class ElasticsearchTest {
...
+        container.start();
+
+        var port9200 = container.getMappedPort(9200);
+        var host = container.getHost();
+        var httpHostAddress = host + ":" + port9200;
```
Comment (sondrev) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:77]: port9200 should be renamed to just port, or a name exaplining what the value means - not what it is. 
I see you also do the same thing in canConnectToContainer() - perhaps you can extract both to a function? like:
`getContainerAddress(container)`

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java [hunk 5]
```diff
@@ -47,41 +49,83 @@ public class ElasticsearchTest {
...
+
+        HttpClient httpClient = HttpClient.newBuilder().build();
+
+        String authenticatorValue = "admin:admin";
+        String credentials = Base64.getEncoder().encodeToString(authenticatorValue.getBytes(StandardCharsets.UTF_8));
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .GET()
+                .uri(URI.create("http://" + httpHostAddress))
+                .header("Authorization", "Basic " + credentials)
```
Comment (sondrev) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java:107]: Do we really need authentication? Dident you turn it off with
`plugins.security.disabled` ?
How is it working in the other tests?

**Review by sondrev — CHANGES_REQUESTED** [review]
Look good. just a few minors

File: search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java [hunk 6]
```diff
@@ -47,41 +49,76 @@ public class ElasticsearchTest {
...
+    @Test
+    void canConnectToContainer() throws IOException, InterruptedException{
+
+        var httpHostAddress = getContainerHost();
+
+        HttpClient httpClient = HttpClient.newBuilder().build();
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .GET()
+                .uri(URI.create("http://" + httpHostAddress))
```
Comment (sondrev) [inline search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java:100]: At this point http:// should just be included in getContainerHost, or getContainerHost should return a URI

**Review by sondrev — APPROVED** [review]
Can you change the variables to be vars?

File: search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java (hunk 6 above)
Comment (fredrikpf) [reply search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java:100]: getContainerHost() is used in setup as well. In the setup we dont expect an URI.

**Pull request merged**